docker/Makefile

```diff
@@ -0,0 +1,73 @@
+SHELL := bash
+.ONESHELL:
+.SHELLFLAGS := -eu -o pipefail -c
+.DELETE_ON_ERROR:
+MAKEFLAGS += --warn-undefined-variables
+MAKEFLAGS += --no-builtin-rules
+
+
+SECRETS_DIR := .secrets
+# File-based recipes
+
+common-secrets: $(SECRETS_DIR)/db_password $(SECRETS_DIR)/db_name $(SECRETS_DIR)/db_user $(SECRETS_DIR)/pepper $(SECRETS_DIR)/file_encryption_key $(SECRETS_DIR)/authjwt_secret_key $(SECRETS_DIR)/organization_name
+.PHONY: common-secrets
+
+secrets-with-username-auth: common-secrets $(SECRETS_DIR)/avatar_first_user_password $(SECRETS_DIR)/avatar_first_user_name ## Initialize secrets for username authentication
+.PHONY: secrets-with-username-auth
+
+secrets-with-email-auth: common-secrets $(SECRETS_DIR)/admin_emails $(SECRETS_DIR)/aws_mail_account_access_key_id $(SECRETS_DIR)/aws_mail_account_secret_access_key ## Initialize secrets for email authentication
+.PHONY: secrets-with-email-auth
+
+
+$(SECRETS_DIR)/db_password:
+	mkdir -p $(SECRETS_DIR)
+	python3 -c "import secrets; print(secrets.token_hex())" > $@
+
+$(SECRETS_DIR)/db_user:
+	mkdir -p $(SECRETS_DIR)
+	touch $@
```

Comment:
Note that I didn't want to have a default value here, so I just used `touch` and not `echo ""` which might have created a database user with an empty name (not sure)